homeassistant/components/screenlogic/__init__.py

```diff
@@ -125,7 +133,15 @@ async def _async_update_data(self):
             async with self.api_lock:
                 await self.hass.async_add_executor_job(self.gateway.update)
         except ScreenLogicError as error:
-            raise UpdateFailed(error) from error
+            _LOGGER.warning("ScreenLogicError - attempting reconnect: %s", error)
+
+            connect_info = get_connect_info(self.hass, self.config_entry)
+            try:
+                self.gateway = ScreenLogicGateway(**connect_info)
```

Comment:
Unfortunately there are side effects when instantiating the gateway so we need to schedule that on the executor too.

Reply:
I'm not sure how to address this, how would I schedule that on the executor?  

My best guess would be create an async function inside `ScreenlogicDataUpdateCoordinator` that instantiates the gateway and saves the reference and call that with `async_add_executor_job`. Is that correct?

Reply:
We can use `functions.partial` to be able to pass keyword arguments and then pass the partial call to `Hass.async_add_executor_job`.

Reply:
i've posted an update which I think satisfies this requirement, but i'm not very well versed in the internals of HASS hopefully I didnt mess it up ;-)